client/client.go

```diff
@@ -210,7 +178,11 @@ func (client *AppConfigClient) send(label string, key string, additionalDecorato
 		return nil, UnexpectedError.wrap(err)
 	}
 
-	resp, err := client.Send(req)
+	var retryDecorator autorest.SendDecorator
+	if retry {
+		retryDecorator = autorest.DoRetryForAttempts(5, 1*time.Second)
+	}
+	resp, err := client.Send(req, retryDecorator)

```

Comment:
@AdamCoulterOz  adding the decorator make the tests panic (`invalid memory address or nil pointer dereference`) in autorest.DecorateSender, but only on the second request. For example, when running a single test, the test method succeeds but the teardown fails.

Reply:
Ok, I've found the issue. We shouldn't pass the decorator if it's not defined